src/sap/fhir/model/r4/FHIRListBinding.js

```diff
@@ -174,14 +174,29 @@ sap.ui.define([
 				this._submitRequest("/ValueSet/$expand", mParameters, fnSuccessValueSet);
 			} else if (!this.aSortersCache && !this.aFilterCache && this.sNextLink && iLength > this.iLastLength) {
 				this.iLastLength += this.iLength;
-				this._submitRequest(this.sNextLink, undefined, fnSuccess, true);
+				// if secure search is enabled convert the next link so that the subsequent calls are handled appropriately
+				if (this.oModel.isSecureSearchModeEnabled() && this.sNextLink && this.sNextLink.indexOf("?") > -1) {
+					var sQueryParams = this.sNextLink.substring(this.sNextLink.indexOf("?") + 1, this.sNextLink.length);
+					var aParameter = sQueryParams ? sQueryParams.split("&") : [];
+					var aKeyValue;
+					for (var i = 0; i < aParameter.length; i++) {
+						aKeyValue = aParameter[i].split("=");
+						mParameters.urlParameters[aKeyValue[0]] = aKeyValue[1];
+					}
+					this._submitRequest(this.sPath, mParameters, fnSuccess, true);
+				} else {
+					this._submitRequest(this.sNextLink, undefined, fnSuccess, true);
+				}
 			} else if (this.iTotalLength === undefined){ // load context based resources
 				this.iLastLength = this.iLength;
 				this._loadProfile(fnSuccessStructDef, fnLoadResources, fnSuccess, mParameters, iLength);
 			} else if (!this._isValueSet()){
 				if (iLength > this.iLastLength){
 					this.iLastLength += this.iLength;
 				}
+				if (!this.iLastLength){
```

Comment:
why this condition is necessary for the secure search feature?

Reply:
this is a bug fix for pagination. I found it while testing the pagination related changes

Reply:
ok....so its a general issue with pagination? Should we create a separate PR for that?

Reply:
yes.. i believe its a general issue with pagination. i am fine with keeping it here as well.. anyways this will be a major release and not a breaking change.

Reply:
what do you suggest ?

Reply:
changelog-whise it would be better to make a separate change. however, with the current release process we create a new release... thats not so nice. we could talk about a better process. but for now leave it like it is.

Reply:
ok